materialize-databricks/driver.go

```diff
@@ -0,0 +1,665 @@
+		{&b.dropStoreSQL, tplDropStore},
+		{&b.mergeInto, tplMergeInto},
+	} {
+		var err error
+    var shardRange = fmt.Sprintf("%08x_%08x", _range.KeyBegin, _range.RClockBegin)
+		if *m.sql, err = RenderTable(target, b.rootStagingPath, shardRange, m.tpl); err != nil {
+			return err
+		}
+	}
+
+	t.bindings = append(t.bindings, b)
+
+	// Drop existing temp tables
+	if _, err := t.load.conn.ExecContext(ctx, b.dropLoadSQL); err != nil {
+		return fmt.Errorf("Exec(%s): %w", b.dropLoadSQL, err)
+	}
+	if _, err := t.store.conn.ExecContext(ctx, b.dropStoreSQL); err != nil {
+		return fmt.Errorf("Exec(%s): %w", b.dropStoreSQL, err)
+	}
+
+	return nil
+}
+
+func (d *transactor) Load(it *pm.LoadIterator, loaded func(int, json.RawMessage) error) error {
+	var ctx = d.Context(it.Context())
+
+  for _, b := range d.bindings {
+    // Create a binding-scoped temporary table for staged keys to load.
+    if _, err := d.load.conn.ExecContext(ctx, b.createLoadTableSQL); err != nil {
+      return fmt.Errorf("Exec(%s): %w", b.createLoadTableSQL, err)
+    }
+  }
+
+  log.Info("load: starting upload of files")
+	for it.Next() {
+		var b = d.bindings[it.Binding]
+
+		b.loadFile.start(ctx)
+
+		if converted, err := b.target.ConvertKey(it.Key); err != nil {
+			return fmt.Errorf("converting Load key: %w", err)
+		} else if err := b.loadFile.encodeRow(converted); err != nil {
+			return fmt.Errorf("encoding row for load: %w", err)
+		}
+	}
+
+  // Copy the staged files in parallel
+  group, groupCtx := errgroup.WithContext(ctx)
+	for idx, b := range d.bindings {
+    if !b.loadFile.started {
+      continue
+    }
+
+    // This is due to a bug in Golang, see https://go.dev/blog/loopvar-preview
+    var bindingCopy = b
+    var idxCopy = idx
+    group.Go(func() error {
+      toCopy, toDelete, err := bindingCopy.loadFile.flush()
+      if err != nil {
+        return fmt.Errorf("flushing load file for binding[%d]: %w", idxCopy, err)
+      }
+      defer d.deleteFiles(groupCtx, toDelete)
+
+      // COPY INTO temporary load table from staged files
+      if _, err := d.load.conn.ExecContext(groupCtx, renderWithFiles(bindingCopy.copyIntoLoad, toCopy...)); err != nil {
+        return fmt.Errorf("load: writing keys: %w", err)
+      }
+
+      return nil
+    })
+	}
+
+  log.Info("load: finished upload of files")
+
+	if it.Err() != nil {
+		return it.Err()
+	}
+
+  if err := group.Wait(); err != nil {
+    return err
+  }
+
+  log.Info("load: starting join query")
+	// Issue a union join of the target tables and their (now staged) load keys,
+	// and send results to the |loaded| callback.
+	rows, err := d.load.conn.QueryContext(ctx, d.load.unionSQL)
+	if err != nil {
+		return fmt.Errorf("querying Load documents: %w", err)
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var binding int
+		var document string
+
+		if err = rows.Scan(&binding, &document); err != nil {
+			return fmt.Errorf("scanning Load document: %w", err)
+		} else if binding > -1 {
```

Comment:
Oh i see it's from the load query if the document field is omitted. The per-transaction query construction would simplify this, since a binding with the document field omitted can't ever get load requests.